garden_ai/app/hpc_notebook.py

```diff
@@ -13,9 +20,62 @@ def hpc_notebook():
 
 
 @hpc_notebook_app.command()
-def start():
+def start(
+    working_directory: str = typer.Option(
+        ..., help="Working directory for the operation."
+    ),
+    notebooks_dir: str = typer.Option(..., help="Directory to bind for notebooks."),
+    container_image: str = "hpc-notebook.sif",
+    definition_file: str = "Singularity.def",
+):
     """Open a notebook file in HPC."""
-    print("🚧🌱🚧 Under Construction 🚧🌱🚧")
+
+    tmp_dir = os.path.abspath(os.path.join(working_directory, "tmp"))
+    notebooks_dir = os.path.abspath(notebooks_dir)
+
+    # Step 1: Create temporary directory if it doesn't exist
+    os.makedirs(tmp_dir, exist_ok=True)
+
+    # Step 2: Set environment variables
+    os.environ["SINGULARITY_TMPDIR"] = tmp_dir
+    os.environ["APPTAINER_TMPDIR"] = tmp_dir
+    os.environ["JUPYTER_RUNTIME_DIR"] = tmp_dir
+    os.environ["JUPYTER_DATA_DIR"] = tmp_dir
+    os.environ["JUPYTER_CONFIG_DIR"] = tmp_dir
+
+    try:
+        # Ensure the definition file exists
+        if not os.path.isfile(definition_file):
+            logger.error(f"Definition file {definition_file} not found.")
```

Comment:
Does this approach rely on the user being in the root directory of the project where `Singularity.def` is? Let's make sure the user can run this command in any directory.

Reply:
Fixed